lambda/matching.ml

```diff
@@ -3100,6 +3257,50 @@ let rec compile_match repr partial ctx (m : pattern_matching) =
       (bind_check str v arg lam, total)
   | _ -> assert false
 
+and compile_simplified repr partial ctx (m : Simple.clause pattern_matching) =
+  match m with
+  | { cases = []; args = [] } -> comp_exit ctx m
+  | { args = ((Lvar v as arg), str) :: argl } ->
+      (* Not sure we need to do anything with arg/v *)
+      let first_match, rem =
+        split_and_precompile_simplified { m with args = (arg, Alias) :: argl }
+      in
+      let lam, total =
+        comp_match_handlers
+          (( if dbg then
+             do_compile_matching_pr
+           else
+             do_compile_matching
+           )
+             repr)
+          partial ctx first_match rem
+      in
+      (bind_check str v arg lam, total)
+  | _ -> assert false
+
+and compile_half_compiled repr partial ctx
+    (m : pattern Non_empty_clause.t pattern_matching) =
+  match m with
+  | { cases = []; args = [] } -> comp_exit ctx m
+  | { args = ((Lvar v as arg), str) :: argl } ->
```

Comment:
This also a new precondition, but `compile_half_compiled` is only called by `compile_flattened`, which is called by `do_for_multiple_match` on a argument list that respect this precondition too. (I have a series of commit at https://github.com/Octachron/ocaml/commits/argo_analysis that makes this point explicit).

Reply:
> I have a series of commit at https://github.com/Octachron/ocaml/commits/argo_analysis that makes this point explicit

Thanks, but we're not going to integrate that right now.
Let's revisit once all the other PRs have been reviewed.